File: 55-jump-game/55-jump-game.java
class Solution {
    
    public boolean canJump(int[] nums) {
        int maxCheck = nums.length-1;
        for(int i=nums.length-2;i>=0;i--){
            if(nums[i]+i>=maxCheck){
                maxCheck = i;    
            }
        }
        return maxCheck==0;
    }
}

File: 55-jump-game/55-jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        
        currentMax = 0
        for i in range(len(nums)):
            currentMax = max(currentMax, i+nums[i])
            if currentMax>=len(nums)-1:
                return True
            if nums[i]==0 and currentMax==i:
                return False
        return False
            
        